### PR TITLE
v3.31.0

### PR DESCRIPTION
**Pull request by TheTrunk**
- Flux App Specification v6
  - expire: v6 brings new parameter of 'expire' determining how many blocks application can live on the network. From 5000 blocks till 264000 blocks (~1 week - ~1 year). This parameter can be adjusted on a 1000 block interval (~1+ day).
  - Default supported options by Home UI are: 1 week, 2 weeks, 1 month, 3 months, 6 months, 1 year.

General:
- Flux Foundation Application address changes from single signature to a 3of5 multisignature address consisting of following core team key holders: Daniel, Parker, Tadeas, Jeremy, Valter. This adds additional trust and security to FluxOS
- Ability of soft forks: 
  - Foundation multisig address can initiate soft-forks of FluxOS by sending proper transactions to itself with soft-fork specifications included in the message field.
  - Current soft fork supported is application price adjustments with more potential flags to come.
  - Soft forks activate immediately on the following block on which transaction was mined.
  - This introduces new database 'chainparams'
- Application prices can now be adjusted on the fly with soft-forks
- Blacklist was extended to also support blacklist of ZelID and application hash
- Continuous better check of blacklisted app images, ZelIDs, and application hashes. This adds faster response of Flux team to fight malicious applications on the network
- Test Launch: Applications can now be test launched. Upon broadcasting a message on the register application section, user can opt in for 'test launch'. This will launch the submitted application specifications on the locally connected stratus node and so app deployer can quickly see how app will perform and catch potential mistakes in set up. Test Launched app live on the node only for a few hours. Only one test launch is allowed per registration.
- Node owner can now launch any application on the global network as well as any prior existing application specification on his node. This launch is temporary and lasts about 36 hours.
- Applications can now have up to 10 components instead of 5
- Every application on flux now has its own flux docker network living on 172.50+X.0.0/16 subnet with 172.50+X.0.1 gateway. This adds more security and separates different apps from being able to talk to each other. X represents the total number of unique apps installed on a specific node. It is possible that 1st application instances will have a network of 172.55.0.1, and 2nd will have 172.51.0.1 in case there are already some apps installed on those nodes. The docker networks created have a format of `fluxDockerNetwork_APPNAME`
- Experimental: Multiple volumes:
  - Application component can now specify multiple paths to mount to the same volume or a volume of a previous component. containerData volumes are separated with `|` character.
  - Eg. Second component containerData `s:/my/data/path|0:/my/another/data|0:/my/anotherB/data|1:/my/path` Second component has its main data volume created and mounted to its location of 'my/data/path' and also specifies flag of s for syncing this data volume. Additionally mounts 3 more volumes - /my/another/data and /my/anotherB/data will be mounted to 1st component data volume, /my/path will be mounted to this second component data volume.
- Autmoatic addition of incoming connections. Flux will now ask other nodes to try initiate connection to our node.
- Additional UI and Backend improvements, fixes.

- v6 specifications activate at blockheight 1,300,000. ALL nodes must update to the release prior this blockheight.

### CODE REVIEW

File: HomeUI/src/views/apps/RegisterFluxApp.vue [hunk 1]
```diff
@@ -1368,6 +1449,42 @@ export default {
           },
         ],
       },
+      appRegistrationSpecificationv6template: {
```
Comment (johnhanlon86) [inline HomeUI/src/views/apps/RegisterFluxApp.vue:1452]:
```suggestion
      appRegistrationSpecificationV6Template: {
```

File: HomeUI/src/views/apps/RegisterFluxApp.vue [hunk 2]
```diff
@@ -1520,14 +1716,22 @@ export default {
...
         this.specificationVersion = 5;
         this.appRegistrationSpecification = this.appRegistrationSpecificationv5template;
         this.appRegistrationSpecification.compose.forEach((component) => {
           const ports = this.getRandomPort();
           // eslint-disable-next-line no-param-reassign
           component.ports = ports;
         });
+      } else {
+        this.specificationVersion = 6;
+        this.appRegistrationSpecification = this.appRegistrationSpecificationv6template;
```
Comment (johnhanlon86) [inline HomeUI/src/views/apps/RegisterFluxApp.vue:1729]:
```suggestion
        this.appRegistrationSpecification = this.appRegistrationSpecificationV6Template;
```

File: ZelBack/src/services/explorerService.js [hunk 3]
```diff
@@ -502,41 +561,49 @@ async function processStandard(blockDataVerbose, database) {
...
+          } catch (error) {
+            log.error(`Hash ${message} already exists. Not adding at height ${blockDataVerbose.height}`);
+            log.error(error);
           }
-        } catch (error) {
-          log.error(`Hash ${message} already exists. Not adding at height ${blockDataVerbose.height}`);
-          log.error(error);
         }
       }
+      // check fo softForks
```
Comment (johnhanlon86) [inline ZelBack/src/services/explorerService.js:602]:
```suggestion
      // check for softForks
```

**Comment by Cabecinha84** [comment]
@TheTrunk is there a way to disable codedev?
With all this spam -> 
"Codecov
/ codecov/patch
ZelBack/src/services/explorerService.js#L904-L905
Added lines #L904 - L905 were not covered by tests"
This is a nightmare to properly review.

File: ZelBack/config/default.js [hunk 4]
```diff
@@ -101,8 +108,10 @@ module.exports = {
       3: 983000, // blockheight. Since this blockheight specification of type 3 is active. User can still submit v1 or v2. UI allows only v2, v3
       4: 1004000, // v4 available, composition
       5: 1142000, // v5 available adding contacts, geolocation
+      6: 1284750, // v6, expiration, app price, t3
     },
     address: 't1LUs6quf7TB2zVZmexqPQdnqmrFMGZGjV6',
+    addressMultisig: 't1UPSwfMYLe18ezbCqnR5QgdJGznzCUYHkj',
```
Comment (Cabecinha84) [inline ZelBack/config/default.js:114]: @TheTrunk Isn't the address multisig a T3 address?

**Review by Cabecinha84 — APPROVED** [review]
Ack

File: ZelBack/config/default.js (hunk 4 above)
Comment (Cabecinha84) [reply ZelBack/config/default.js:114]: t3aGJvdtd8NR6GrnqnRuVEzH6MbrXuJFLUX ?

Comment (TheTrunk) [reply ZelBack/config/default.js:114]: Correct as well as blockheight. I have adjusted those temporarily for testing purposes

**Pull request merged**